### ISSUE TITLE
TypeScript HasOne association create requires foreign key that is automatically included

### ISSUE THREAD

**Issue by garrett-gottlieb** [sequelize/sequelize#14365]
## Issue Creation Checklist

- [X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### SSCCE

```js
class Subscription extends Model<
  InferAttributes<Subscription>,
  InferCreationAttributes<Subscription>
> {
  declare id: CreationOptional<Scalars['ID']>
  declare createdAt: CreationOptional<Date>
  declare updatedAt: CreationOptional<Date>

  declare userId: Scalars['ID']

  declare iosTransactionReceipt: string
  declare isActive: boolean

  declare readonly user?: User

  declare static associations: {
    user: Association<Subscription, User>
  }
}
Subscription.init(
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    createdAt: DataTypes.DATE,
    updatedAt: DataTypes.DATE,
    iosTransactionReceipt: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    isActive: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
    },
    userId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      unique: true,
    },
  },
  {
    sequelize: db,
  }
)

// Usage
user.createSubscription({
      isActive: true,
      iosTransactionReceipt,
    })
```

### What do you expect to happen?

No TS error

### What is actually happening?

```
Argument of type '{ isActive: true; iosTransactionReceipt: string; }' is not assignable to parameter of type 'Optional<InferCreationAttributes<Subscription, { omit: never; }>, NullishPropertiesOf<InferCreationAttributes<Subscription, { omit: never; }>>>'.
  Property 'userId' is missing in type '{ isActive: true; iosTransactionReceipt: string; }' but required in type 'Omit<InferCreationAttributes<Subscription, { omit: never; }>, NullishPropertiesOf<InferCreationAttributes<Subscription, { omit: never; }>>>'.ts(2345)
```

```
Output here
```

### Additional context

Add any other context and details here.

### Environment

- Sequelize version: 6.18
- Node.js version:
- If TypeScript related: TypeScript version: 4.4
- Database & Version:
- Connector library & Version:

## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
A new version of Sequelize 6 is being released with a backport of https://github.com/sequelize/sequelize/pull/14148, which should solve your issue

This is the relevant documentation to resolve your issue: https://sequelize.org/docs/v6/other-topics/typescript/#the-case-of-modelinit

TL;DR your class should look like this

```typescript
class Subscription extends Model<
  InferAttributes<Subscription>,
  InferCreationAttributes<Subscription>
> {
  declare id: CreationOptional<Scalars['ID']>
  declare createdAt: CreationOptional<Date>
  declare updatedAt: CreationOptional<Date>

  declare userId: ForeignKey<Scalars['ID']> // I changed this line

  declare iosTransactionReceipt: string
  declare isActive: boolean

  declare readonly user?: User

  declare static associations: {
    user: Association<Subscription, User>
  }
}
```
